text/0000-security-process.md

```diff
@@ -0,0 +1,258 @@
+do and the complex blockchain related things we are doing today that we want
+formally verified. This may be more viable in the future, but it would require
+significant time and resource investment. It would not be responsible to rely
+on formal verification for a project like Grin today.
+
+#### What is the impact of not doing this?
+
+If we do not adopt a standard as proposed here, the community is left with
+unresolved problems:
+
+- Who do we tell if we receive a vulnerability?
+- What details are we obligated to share?
+    - Exploit code?
+    - What if it is an inflation bug?
+- Do we attempt to collect a bug bounty?
+
+The community is exposed to risk by not having clear answers to these questions
+_before_ it is presented with a critical security vulnerability. Even if the
+vulnerability has technically been fixed, there may be severe fallout in the
+community if it perceives the situation wasn't handled ethically by the parties
+involved in the disclosure. There is also risk of not implementing a fix in time
+across the community because there were internal disputes about who to tell.
+
+Without a public pre-commitment to expected actions taken during a vulnerability
+disclosure, Grin runs the risk of alienating the community in pursuit of
+security, or even worse, not implementing a critical fix before any user
+experiences privacy or value loss.
+
+# Prior art
+[prior-art]: #prior-art
+
+The traditional disclosure model (RF Policy/security@) [0] handles interaction between
+researcher and vendor but does not quite fit for our use case of potentially multiple
+vendors up/down stream. There have been multi-party models proposed with single
+researcher and multiple vendors that have not been successful, nor do they
+sufficiently account for the challenges faced in public decentralized projects.
+
+Neither model fits with chains of vendors sharing code, protocols with multiple
+implementations and forks, all operating in public. We need a model that is
+better suited to handle cases of consensus critical vulnerabilities that require
+upgrades across many implementations, forks and projects both in the Grin and
+greater cryptocurrency ecosystems.
+
+Bitcoin has received some criticism over handling of disclosures, as it never
+had a robust, well-defined pre-committed standard to follow. We want to learn
+from those lessons to avoid putting more people in the position to repeat the
+same mistakes.
+
+Zcash, a privacy based cryptocurrency, experienced a critical vulnerability
+that inspired the creation of the standard this RFC proposes to adopt [1]. Due
+to the similar nature of the Zcash and Grin ecosystems, this prior art is highly
+relevant to the security process for Grin.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- How can we further reduce centralization risk for Grin's security process?
+
+- How can we responsibly handle vulnerabilities in academic papers?
```

Comment:
What do you mean here?

Reply:
Updated some details for those questions in https://github.com/mimblewimble/grin-rfcs/pull/13/commits/6b0aee02c7d5db79a140380c5011374df842737b. 

> How can we further reduce centralization risk for Grin's security process?

Essentially there are always ways to further reduce centralization risk, it is never a complete process.

>How can we responsibly handle vulnerabilities in academic papers?

A vulnerability could be discovered at specification level in an assumption in a paper we rely on for security etc. That is a bit of a trickier disclosure and not as straightforward as updating code in a repo or adding a banner on a website. 

Thanks very much for reviewing, let me know if you'd like more clarification here.

Reply:
Okay got it thanks for the clarification!